crates/wasm-encoder/src/core/types.rs

```diff
@@ -347,30 +347,46 @@ impl RefType {
     /// Alias for the `funcref` type in WebAssembly
     pub const FUNCREF: RefType = RefType {
         nullable: true,
-        heap_type: HeapType::Func,
+        heap_type: HeapType::Abstract {
+            shared: false,
+            ty: AbstractHeapType::Func,
+        },
```

Comment:
If we think this is too wordy, I'm open to suggestions: I considered a helper like `HeapType::abstract(false, AbstractHeapType::func)` but decided to just directly construct this explicitly everywhere. But it's an option...

Reply:
I think this is fine, it's just "the one place" to write this down for the `pub const` values